### DISCUSSION TITLE
Finding a good layout - General Diagnosis Room

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2406] [General Diagnostics]
I am exploring good design layouts for each room, in this post I am focusing on the General Diagnosis room.
**The Goal**
Keeping the room as small as possible while keeping the doctor as happy as possible(avoid being unhappy and wanting a pay rise) while optimizing turnaround time for handyman plant watering and patient walk times.

**Version 1.00**
![Screenshot 2023-09-21 124654](https://github.com/CorsixTH/CorsixTH/assets/13173133/078abc21-36a4-4e6c-8219-fd9be07d5e02)

**Items**
1x Crash Trolley
1x Screen
1x Plant
1x Bin
2x Radiator
1x Fire Extinguisher

**Summary**
 The main aspect for the General Diagnosis room is patient walk times, we want the patient to change as fast as possible behind the screen and then move as fast as possible to the crash trolley.
In this design the patient does not need to walk around the screen, they just walk in turn left and change. While walking to the screen they also pass the plant and bin. After they change they move north to the crash trolley with very minimal walking.

Notes
1. Is the screen in the best position? - I think so as the patient does not need to make any turns while walking to the screen and it is as close to the door as possible.
2. Is the plant in the best position? - For this design yes, it is as close to the door as possible while not affecting the patient walk path to the screen.
3. Is the bin in the best position? - For this design yes, it is along the path for the patient so they can use it while not affecting the patient walk path to the screen
4. Is the fire extinguisher in the best position? - For now i am unsure (do we need more? does it need to be closer to the Crash Trolley? is it even doing anything for this room?)
5. Are the Radiators in the best position? - They do have a good heat map, i am not sure if more would be better.
![Screenshot 2023-09-23 093640](https://github.com/CorsixTH/CorsixTH/assets/13173133/52ce4498-9a8d-4a56-b99e-723466cf7fa4)
6. Is the crash trolley in the best position? - This is where a bit of research could be done, from my testing it is quite in a good spot as the patient does not need to walk much to get to it.

Lets start the decision, can this design be improved and if so how?

**Version 1.01**
Updating fire extinguisher placing (with it having a 2 tile radius), now the best spot would be the center of the room so 1 fire extinguisher could cover the whole room however the Crash Trolley and Screen are in the way. To solve this we can use 2 fire extinguishers in the room.
![Screenshot 2023-09-24 095711](https://github.com/CorsixTH/CorsixTH/assets/13173133/f87b939e-cbfa-4ea7-8f9d-97e8b7d70ca0)
I have also moved the Radiators so the fire extinguishers would fit.

**Version 1.02**
![Screenshot 2023-09-25 103132](https://github.com/CorsixTH/CorsixTH/assets/13173133/370cf474-6e10-460e-9e29-7c62c830fe72)
Changed door to screen position, now patient walks 1 square into the screen with no turns and then they walk one square to Crash Trolley no turns.
Plant is still by the door.
Bin is in the path of the patient.
Fire Extinguishers should cover the whole room (2 square radius)

**Comment by PaveMentman** [comment]
---
"General Diagnosis"-Room pretty is pretty much identical when it comes to patient-walkpath-routine and thus room-planning as "Cardigram Room" did;
I won't repeat the very same things I already mentioned here:
https://github.com/CorsixTH/CorsixTH/discussions/2407#discussioncomment-7093533

However, the important reminders (to myself at least):
- Screen cannot be rotated
-> Patients needs to do the change of clothes always the on the North(-West?) / Up(-Left?)-square of the screen.

Therefore, for General Diagnosis
(and any other other room when patient first has to change clothes behind the Screen),
the most optimised entrances are limited to North-walls.


When it comes to Crash Trolley, the patient stands in front the red shelves / cabinets


Here is perhaps my most optimized General Diagnosis-room when the entrance is placed North-East-side 
( again, the radiator-location is "corrected" with the increased thermostat thus the heating reaches all the occupants in the room ):

![2023-09-24-15-54-42-515-CorsixTH-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/6709463/506cd6a1-188a-439e-b7c8-6dcd19aadc32)
![2023-09-24-15-54-45-566-CorsixTH-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/6709463/6d66f470-bb01-4068-b78d-72fd95dc74f5)
---

**Reply by ageekhere** [reply]
Updated Version 1.02 with the door idea, let me know which is better either walking into the screen from the door in my Version 1.02 one or turning into the screen in your one.
